backend/src/crd/controller/scheduledworkflow/util/cron_schedule.go

```diff
@@ -42,79 +46,80 @@ func NewCronSchedule(cronSchedule *swfapi.CronSchedule) *CronSchedule {
 
 // GetNextScheduledEpoch returns the next epoch at which a workflow must be
 // scheduled.
-func (s *CronSchedule) GetNextScheduledEpoch(lastJobEpoch *int64,
-	defaultStartEpoch int64) int64 {
-	effectiveLastJobEpoch := s.getEffectiveLastJobEpoch(lastJobEpoch, defaultStartEpoch)
-	return s.getNextScheduledEpoch(effectiveLastJobEpoch)
+func (s *CronSchedule) GetNextScheduledEpoch(lastJobTime *v1.Time,
+	defaultStartTime time.Time, location *time.Location) int64 {
```

Comment:
nit: I think either taking epoch in epoch out or time in time out. It's not consistent if you take time as input and epoch as output.

I'd suggest keeping epoch as input, because it will be more consistent with `PeriodicSchedule`.

Reply:
I see your point, but in that case we would convert ``` s.Status.Trigger.LastTriggeredTime```(v1.Time) -> ``` *int``` -> ``` time.Time``` which don't make sense to me. In that case I favour to return time.Time instead.

Reply:
Makes sense to me, then we can return time instead

Reply:
We'd need to rename the function too: `GetNextScheduledTime`